server/src/main/java/org/tctalent/server/service/db/impl/SavedSearchServiceImpl.java

```diff
@@ -137,781 +136,777 @@
-        }
-        if (request.getDisplayedFieldsShort() != null) {
-            savedSearch.setDisplayedFieldsShort(request.getDisplayedFieldsShort());
-        }
-        savedSearchRepository.save(savedSearch);
+    savedSearchRepository.save(savedSearch);
+  }
+
+  private Specification<Candidate> computeQuery(
+      SearchCandidateRequest request, @Nullable Collection<Candidate> excludedCandidates) {
+    //There may be no logged in user if the search is called by the
+    //overnight Watcher process.
+    User user = userService.getLoggedInUser();
+
+    //This list is initialized with the main saved search id, but can be
+    //added to by addQuery below when the search is built on other
+    //searches. The idea is to avoid circular dependencies between searches.
+    //For example, in the simplest case we don't want a saved search
+    //to be based on itself.
+    List<Long> searchIds = new ArrayList<>();
+    if (request.getSavedSearchId() != null) {
+      searchIds.add(request.getSavedSearchId());
     }
 
-    private Specification<Candidate> computeQuery(
-        SearchCandidateRequest request, @Nullable Collection<Candidate> excludedCandidates) {
-        //There may be no logged in user if the search is called by the
-        //overnight Watcher process.
-        User user = userService.getLoggedInUser();
-
-        //This list is initialized with the main saved search id, but can be
-        //added to by addQuery below when the search is built on other
-        //searches. The idea is to avoid circular dependencies between searches.
-        //For example, in the simplest case we don't want a saved search
-        //to be based on itself.
-        List<Long> searchIds = new ArrayList<>();
-        if (request.getSavedSearchId() != null) {
-            searchIds.add(request.getSavedSearchId());
-        }
-
-        Specification<Candidate> query = CandidateSpecification
-            .buildSearchQuery(request, user, excludedCandidates);
-        if (org.apache.commons.collections.CollectionUtils.isNotEmpty(request.getSearchJoinRequests())) {
-            for (SearchJoinRequest searchJoinRequest : request.getSearchJoinRequests()) {
-                query = addQuery(query, searchJoinRequest, searchIds);
-            }
-        }
-        return query;
+    Specification<Candidate> query = CandidateSpecification
+        .buildSearchQuery(request, user, excludedCandidates);
+    if (org.apache.commons.collections.CollectionUtils.isNotEmpty(request.getSearchJoinRequests())) {
+      for (SearchJoinRequest searchJoinRequest : request.getSearchJoinRequests()) {
+        query = addQuery(query, searchJoinRequest, searchIds);
+      }
     }
+    return query;
+  }
 
-    private BoolQueryBuilder computeElasticQuery(
-        SearchCandidateRequest request, @Nullable String simpleQueryString,
-        @Nullable Collection<Candidate> excludedCandidates) {
+  private BoolQueryBuilder computeElasticQuery(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 30. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 1038-1119 into a separate method.